src/build/tenscript/shape_phase.rs

```diff
@@ -180,21 +185,22 @@ impl ShapePhase {
                 StartCountdown(DEFAULT_VULCANIZE_COUNTDOWN)
             }
             ShapeOperation::ReplaceFaces => {
+                self.complete_all_shapers(fabric);
                 for face_id in fabric.replace_faces() {
                     fabric.remove_face(face_id);
                 }
                 Noop
             }
-            ShapeOperation::SetViscosity { viscosity } =>
-                SetViscosity(viscosity),
+            ShapeOperation::SetViscosity { viscosity } => SetViscosity(viscosity),
+            ShapeOperation::Bouncy => Bouncy,
         }
     }
 
     fn complete_shaper(&self, fabric: &mut Fabric, Shaper { interval, alpha_face, omega_face, join, .. }: Shaper) {
+        fabric.remove_interval(interval);
         if join {
             fabric.join_faces(alpha_face, omega_face);
         }
-        fabric.remove_interval(interval);
```

Comment:
Ooh, is there a story here? Was something breaking?

Reply:
No.  It actually doesn't matter.